market/deal_ingest_snap.go

```diff
@@ -0,0 +1,572 @@
+				}
+
+				if cn != 1 {
+					return false, xerrors.Errorf("adding sector to pipeline: incorrect number of rows returned")
+				}
+
+				_, err = tx.Exec("SELECT transfer_and_delete_open_piece_snap($1, $2)", p.mid, sector.number)
+				if err != nil {
+					return false, xerrors.Errorf("adding sector to pipeline: %w", err)
+				}
+			}
+
+		}
+		return true, nil
+	}, harmonydb.OptionRetry())
+
+	if err != nil {
+		return xerrors.Errorf("start sealing sector: %w", err)
+	}
+
+	if !comm {
+		return xerrors.Errorf("start sealing sector: commit failed")
+	}
+
+	return nil
+}
+
+func (p *PieceIngesterSnap) AllocatePieceToSector(ctx context.Context, maddr address.Address, piece lpiece.PieceDealInfo, rawSize int64, source url.URL, header http.Header) (api.SectorOffset, error) {
+	if maddr != p.miner {
+		return api.SectorOffset{}, xerrors.Errorf("miner address doesn't match")
+	}
+
+	var psize abi.PaddedPieceSize
+
+	if piece.PieceActivationManifest != nil {
+		psize = piece.PieceActivationManifest.Size
+	} else {
+		psize = piece.DealProposal.PieceSize
+	}
+
+	// check raw size
+	if psize != padreader.PaddedSize(uint64(rawSize)).Padded() {
+		return api.SectorOffset{}, xerrors.Errorf("raw size doesn't match padded piece size")
+	}
+
+	var propJson []byte
+
+	dataHdrJson, err := json.Marshal(header)
+	if err != nil {
+		return api.SectorOffset{}, xerrors.Errorf("json.Marshal(header): %w", err)
+	}
+
+	vd := verifiedDeal{
+		isVerified: false,
+	}
+
+	if piece.DealProposal != nil {
+		// For snap we convert f05 deals to DDO
+
+		alloc, err := p.api.StateGetAllocationIdForPendingDeal(ctx, piece.DealID, types.EmptyTSK)
+		if err != nil {
+			return api.SectorOffset{}, xerrors.Errorf("getting allocation for deal %d: %w", piece.DealID, err)
+		}
+		clid, err := p.api.StateLookupID(ctx, piece.DealProposal.Client, types.EmptyTSK)
+		if err != nil {
+			return api.SectorOffset{}, xerrors.Errorf("getting client address for deal %d: %w", piece.DealID, err)
+		}
+
+		clientId, err := address.IDFromAddress(clid)
+		if err != nil {
+			return api.SectorOffset{}, xerrors.Errorf("getting client address for deal %d: %w", piece.DealID, err)
+		}
+
+		var vac *miner2.VerifiedAllocationKey
+		if alloc != verifreg.NoAllocationID {
+			vac = &miner2.VerifiedAllocationKey{
+				Client: abi.ActorID(clientId),
+				ID:     verifreg13.AllocationId(alloc),
+			}
+		}
+
+		payload, err := cborutil.Dump(piece.DealID)
+		if err != nil {
+			return api.SectorOffset{}, xerrors.Errorf("serializing deal id: %w", err)
+		}
+
+		piece.PieceActivationManifest = &miner.PieceActivationManifest{
+			CID:                   piece.PieceCID(),
+			Size:                  piece.DealProposal.PieceSize,
+			VerifiedAllocationKey: vac,
+			Notify: []miner2.DataActivationNotification{
+				{
+					Address: market.Address,
+					Payload: payload,
+				},
+			},
+		}
+
+		piece.DealProposal = nil
+		piece.DealID = 0
+		piece.PublishCid = nil
+	}
```

Comment:
We should do this conversion just before the ProveUpdate message. We do the same thing for ProveCommit 3. This approach looses important metadata about the deal. Post DDO, metadata is very tricky on chain and deleting this metadata locally is not something I would recommend.

Reply:
Not so clear, I suspect that things will move more and more into the direction of DDO / custom markets, I don't think we need to treat the builtin markets in a glorified way

In the UI we can still decode the PAM and display the DealID